File: src/languages/vanillajs/binary/converter.js
import { BitLimit } from "./config.js"

export function BoolToStr(a){

	let o = "0"
	if (a == true){
		o = "1"
	}	

	return o
}


export function StrToBool(a){

	let o = false
	if (a == "1"){
		o = true
	}

	return o
}

export function CompileString(boolr){

	let ns = new Array(BitLimit)

	for (let i=0; i<BitLimit; i++) {
		ns[i] = BoolToStr(boolr[i])
	}

	return ns.join("")
}


export function DecompileString(a){

	let ns = new Array(a.split("").length)
	for (let i=0; i<a.split("").length; i++) {
		ns[i] = StrToBool(a.split("")[i])
	}
	return ns
}

export function ReverseList(l){
	let e=l.length-1
	let l2=new Array(l.length)
	for (let i=0; i<l.length; i++) {

		l2[e]=l[i]
		e=e-1
	}
	return l2
}

export function FixBinary(bin){
	let a=ReverseList(DecompileString(bin))
	if (BitLimit-a.length < 0){
		console.error("FATAL ERROR: Number Is Larger Than BitLimit "+String(BitLimit))
	}
	const e= a.length
	a.length = BitLimit
	if (BitLimit-e != 0 || BitLimit-e <! 0){
		for (let i=0; i<(BitLimit-(e)); i++) {
			a[i+e]=false
	}
}
	return CompileString(ReverseList(a))
}

File: src/languages/c#/binary/converter.cs
namespace minicallib_converter{

    using System;
    using config;

    public class converter {
        
        public static String BoolToStr(boolean a) {

            String o = "0";
            if (a == true) { o = "1"; }
        
            return o;
        
        }

        public static boolean StrToBool(String a) {

            boolean o = false;
            if (a.contains("1")) { o = true; }
        
            return o;
        
        }
        
        public static String CompileString(params bool[] bools) {
        
            String[] ns = new String[config.BitLimit];
        
            for (int i = 0; i < bools.length; i++) {
        
                ns[i] = BoolToStr(bools[i]);
        
            }
        
            string s = ns.ToString();

            return s;
        
        }
        
        public static boolean[] DecompileString(String a){

            boolean[] ns = new boolean[a.length()];
            for (int i=0; i<a.split("").length; i++) {
                ns[i] = StrToBool(a.split("")[i]);
            }
            return ns;
        }
        
        public static bool[] ReverseList(bool[] l){
            int e=l.length-1;
            boolean[] l2= new boolean[l.length];
            for (int i=0; i<l.length; i++) {
        
                l2[e]=l[i];
                e--;
            }
            return l2;
        }
        
        public static String FixBinary(String bin){
            boolean[] a = ReverseList(DecompileString(bin));
            boolean[] a2 = new boolean[config.BitLimit];
            if (config.BitLimit-a.length < 0){
                Console.WriteLine("FATAL ERROR: Number Is Larger Than BitLimit "+string.Format("%s", config.BitLimit));
            }
            int e= a.length;
            for (int i=0; i<e; i++){
                a2[i] = a[i];
            }
            if (config.BitLimit-e != 0 || config.BitLimit-e > -1){
                for (int i=0; i<(config.BitLimit-(e)); i++) {
                    a2[i+(e)] = false;
                }
            }
            return CompileString(ReverseList(a2));
        }


    }
}